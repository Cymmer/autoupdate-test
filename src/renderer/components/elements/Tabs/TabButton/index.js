import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import styles from '../tabs.module.scss';

import IconButton from '../../Button/IconButton';
import iconButtonTypes from '../../Button/constants/iconButtonTypes';

import tabTypes from '../constants/tabTypes';

const TabButton = ({
  innerRef,
  className,
  children,
  type,
  active,
  onClick,
  closeAction,
  id,
  hasNewData,
}) => (
  <button
    ref={(e) => {
      if (innerRef) {
        innerRef.current = e;
      }
    }}
    className={cn(className, {
      [styles[`Tab___${type}___active`]]: active,
      [styles[`Tab___${type}`]]: !active,
      [styles[`Tab___${type}___hasNewData`]]: hasNewData,
      [styles.Tab___withClose]: closeAction && active,
    })}
    data-test="tabButton"
    id={id}
    type="button"
    onClick={onClick}
  >
    {children}
    {closeAction && active && (
      <IconButton
        className={styles.Tab_close}
        data-test="closeButton"
        icon="close"
        type={iconButtonTypes.SOLID.XS}
        onClick={closeAction}
      />
    )}
  </button>
);

TabButton.defaultProps = {
  className: null,
  active: false,
  type: tabTypes.HORIZONTAL.LG,
  closeAction: null,
  id: null,
  hasNewData: false,
};

TabButton.propTypes = {
  innerRef: PropTypes.object,
  className: PropTypes.string,
  active: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  type: PropTypes.oneOf([
    tabTypes.HORIZONTAL.LG,
    tabTypes.HORIZONTAL.SM,
    tabTypes.VERTICAL.LG,
    tabTypes.VERTICAL.SM,
  ]),
  closeAction: PropTypes.func,
  id: PropTypes.string,
  hasNewData: PropTypes.bool,
};

export default TabButton;
