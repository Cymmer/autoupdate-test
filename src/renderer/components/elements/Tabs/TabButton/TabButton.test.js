import React from 'react';
import { shallow } from 'enzyme';
import { findByTestAttr } from 'codechum-app-utils';
import TabButton from './index';

describe('RENDERING - Tabs > TabButton', () => {
  it('should render with default props', () => {
    const component = shallow(<TabButton onClick={() => {}}>Test</TabButton>);
    expect(component).toMatchSnapshot();
  });

  it('should render the close button if there is a close action and is active', () => {
    const component = shallow(
      <TabButton active closeAction={() => {}} onClick={() => {}}>
        Test
      </TabButton>
    );
    expect(component).toMatchSnapshot();
  });
});

describe('EVENTS - Tabs > TabButton', () => {
  it('should be clickable', () => {
    const clickFn = jest.fn();
    const component = shallow(<TabButton onClick={clickFn}>Test</TabButton>);

    findByTestAttr(component, 'tabButton').simulate('click');

    expect(clickFn).toHaveBeenCalled();
  });

  it('should have a clickable close button if it has a close button', () => {
    const clickFn = jest.fn();
    const component = shallow(
      <TabButton active closeAction={clickFn} onClick={() => {}}>
        Test
      </TabButton>
    );

    findByTestAttr(component, 'closeButton').simulate('click');

    expect(clickFn).toHaveBeenCalled();
  });
});
