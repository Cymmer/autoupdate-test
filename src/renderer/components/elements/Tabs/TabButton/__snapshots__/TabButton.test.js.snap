// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - Tabs > TabButton should render the close button if there is a close action and is active 1`] = `
<button
  className="Tab___horizontal___lg___active Tab___withClose"
  data-test="tabButton"
  id={null}
  onClick={[Function]}
  type="button"
>
  Test
  <IconButton
    className="Tab_close"
    colorName="blue"
    data-test="closeButton"
    disabled={false}
    icon="close"
    iconClassName={null}
    id={null}
    isLocked={false}
    kind="button"
    onClick={[Function]}
    style={null}
    tabIndex={0}
    type="solid___xs"
  />
</button>
`;

exports[`RENDERING - Tabs > TabButton should render with default props 1`] = `
<button
  className="Tab___horizontal___lg"
  data-test="tabButton"
  id={null}
  onClick={[Function]}
  type="button"
>
  Test
</button>
`;
